datasets/code_refinement/code_refinement.py

```diff
@@ -0,0 +1,179 @@
+  title={CodeXGLUE: A Benchmark Dataset and Open Challenge for Code Intelligence},
+  year={2020},
+}
+"""
+
+# TODO: Add description of the dataset here
+# You can copy an official description
+_DESCRIPTION = """\
+Millions of open-source projects with numerous bug fixes
+are available in code repositories. This proliferation
+of software development histories can be leveraged to
+learn how to fix common programming bugs
+Code refinement aims to automatically fix bugs in the code,
+which can contribute to reducing the cost of bug-fixes for developers.
+Given a piece of Java code with bugs,
+the task is to remove the bugs to output the refined code. """
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = "MIT License"
+
+# TODO: Add link to the official dataset URLs here
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URLs = {
+    "code_refinement_small": {
+        "train_fix": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/small/train.buggy-fixed.fixed",
+        "valid_fix": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/small/valid.buggy-fixed.fixed",
+        "test_fix": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/small/test.buggy-fixed.fixed",
+        "train_bug": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/small/train.buggy-fixed.buggy",
+        "valid_bug": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/small/valid.buggy-fixed.buggy",
+        "test_bug": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/small/test.buggy-fixed.buggy",
+    },
+    "code_refinement_medium": {
+        "train_fix": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/medium/train.buggy-fixed.fixed",
+        "valid_fix": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/medium/valid.buggy-fixed.fixed",
+        "test_fix": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/medium/test.buggy-fixed.fixed",
+        "train_bug": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/medium/train.buggy-fixed.buggy",
+        "valid_bug": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/medium/valid.buggy-fixed.buggy",
+        "test_bug": "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Code-Code/code-refinement/data/medium/test.buggy-fixed.buggy",
+    },
+}
+
+
+# TODO: Name of the dataset usually match the script name with CamelCase instead of snake_case
+class NewDataset(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class CodeRefinement(datasets.GeneratorBasedBuilder):
```
this needs to be the camel case version of the dataset script name `code_refinement`